github/rules.go

```diff
@@ -704,7 +709,7 @@ func (r *RepositoryRulesetRules) UnmarshalJSON(data []byte) error {
 			r.RequiredDeployments = &RequiredDeploymentsRuleParameters{}
 
 			if w.Parameters != nil {
-				err = json.Unmarshal(*w.Parameters, r.RequiredDeployments)
+				err = json.Unmarshal(w.Parameters, r.RequiredDeployments)
```

Comment:
Throughout this PR, could you please use the more idiomatic Go construct of `if err := ... ; err != nil {` instead of the two-line equivalent, for consistency with the rest of this repo?

Reply:
Actually I can't see a single example of this pattern in the whole repo... 👀

Reply:
You can't find `if err := ...; err != nil {` anywhere?
I see them:

```
$ rg 'if err :=' | wc -l
      65
```

Reply:
I might have been having a moment this morning, let's blame it on a lack of sleep... :disappointed:

Reply:
Done